src/features/singleRight/components/SearchSection/SearchSection.tsx

```diff
@@ -62,6 +63,10 @@ export const SearchSection = ({ onAdd, onUndo }: SearchSectionParams) => {
     resultsPerPage: searchResultsPerPage,
   });
 
+  const env = getEnvFromUrl();
+  const displayPopularResources =
+    !searchString && filters.length === 0 && (env === Environment.PROD || env === Environment.TT02);
+
```

Comment:
I'm not entirely convinced that this is the best approach. Currently, the logic for turning features on or off is both in the bff and the client. I propose centralizing this logic to make it easier to manage and maintain. 

One potential solution is to pass the feature flags from the backend for frontend (BFF) in the `window` object. This would allow us to access them directly from the client-side code.

We can achieve this by:

1. **Create a `featureFlags` object:**

   ```csharp
   if (await ShouldShowAppView())
   {
       ViewBag.featureFlags = _logicFlags; // _logicFlags or feature_flags?
       return View();
   }
   ```

2. **Serialize and embed in the view:**

   ```cshtml
   <body>
     <script type="text/javascript">
       window.featureFlags = @Html.Raw(Json.Serialize(ViewBag.featureFlags));
     </script>
     <div id="root"></div>
   ...
   </body>
   ```

3. **Use the feature flags in js:**

   ```javascript
   const displayPopularResources = window.featureFlags.displayPopularResources;
   ```

I believe this approach provides better maintainability and separation of concerns. However, I'm curious to hear what you think.

Reply:
This is so much better!
I absolutely agree with you that one single source of truth is better than two and for the BFF to share it's flags with the frontend is a great way to go about it (that I wasn't aware could be done like this).
Thank you for this excellent feedback! 😄🙌